code/modules/research/protolathe.dm

```diff
@@ -202,3 +202,16 @@
 			if(new_item.matter && new_item.matter.len > 0)
 				for(var/i in new_item.matter)
 					new_item.matter[i] = new_item.matter[i] * mat_efficiency
+
+/obj/machinery/r_n_d/protolathe/proc/eject(var/material, var/amount)

```

Comment:
Just.. move the vars that need to be shared down to r_n_d? If they're identical in structure and purpose then I don't see an issue.